src/pgwire/src/message.rs

```diff
@@ -362,6 +362,7 @@ impl ErrorResponse {
             AdapterError::SqlCatalog(_) => SqlState::INTERNAL_ERROR,
             AdapterError::SubscribeOnlyTransaction => SqlState::INVALID_TRANSACTION_STATE,
             AdapterError::Transform(_) => SqlState::INTERNAL_ERROR,
+            AdapterError::UnallowedOnCluster { .. } => SqlState::INTERNAL_ERROR,
```

Comment:
This doesn't seem like an `INTERNAL_ERROR` to me. I'm not sure what the best error code to use here is since this scenario is not possible in PostgreSQL. Maybe `SqlState::INSUFFICIENT_PRIVILEGE`? Though that one maybe implies there's some privilege that would allow the user to execute this. Maybe `SqlState::S_R_E_PROHIBITED_SQL_STATEMENT_ATTEMPTED`?

Whatever you pick it should indicate that the user did something they weren't allowed to, not that it was an internal error. (For reference: https://www.postgresql.org/docs/current/errcodes-appendix.html).

Reply:
That makes sense to me! I updated it to be `SqlState::S_R_E_PROHIBITED_SQL_STATEMENT_ATTEMPTED`, nothing else seemed to be a great fit. Maaaaaybe `INSUFFICIENT_RESOURCES` but that's not really correct either